index.js

```diff
@@ -184,12 +185,12 @@ function monkeypatch() {
     }
     if (node.typeParameters) {
       for (var j = 0; j < node.typeParameters.params.length; j++) {
-        checkIdentifierOrVisit.call(this, node.typeParameters.params[j]);
+        checkIdentifierOrVisit.call(this, node.typeParameters.params[j], true);
       }
     }
     if (node.superTypeParameters) {
       for (var k = 0; k < node.superTypeParameters.params.length; k++) {
-        checkIdentifierOrVisit.call(this, node.superTypeParameters.params[k]);
+        checkIdentifierOrVisit.call(this, node.superTypeParameters.params[k], true);
```

Comment:
superTypeParams should pass `, false` here, right? They're not defining a type param, but using one (and might actually use something else:

```
class A<T> extends B<T, number> { ... }
```

Here the `node.typeParameters` would be `T` which we should define in this new scope, and `node.superTypeParameters` would be `T, number` which we should check to see that they are defined in the scope.